src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java

```diff
@@ -616,36 +619,34 @@ public Runnable lockStudentParticipations(ProgrammingExercise exercise, Predicat
+            groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise,
                     Constants.PROGRAMMING_EXERCISE_FAILED_LOCK_OPERATIONS_NOTIFICATION + numberOfFailedLockOperations);
         }
         else {
-            groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise.get(),
-                    Constants.PROGRAMMING_EXERCISE_SUCCESSFUL_LOCK_OPERATION_NOTIFICATION);
+            groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise, Constants.PROGRAMMING_EXERCISE_SUCCESSFUL_LOCK_OPERATION_NOTIFICATION);
         }
 
         // Stash the not submitted/committed changes for exercises with manual assessment and with online editor enabled
         // This is necessary for students who have used the online editor, to ensure that only submitted/committed changes are displayed during manual assessment
         // in the case they still have saved changes on the Artemis server which have not been committed / pushed
         // NOTE: we always stash, also when manual assessment is not activated, because instructors might change this after the exam
         if (Boolean.TRUE.equals(exercise.isAllowOnlineEditor())) {
-            List<ProgrammingExerciseStudentParticipation> failedStashOperations = stashChangesInAllStudentRepositories(programmingExerciseId, condition);
-            long numberOfFailedStashOperations = failedStashOperations.size();
-            if (numberOfFailedStashOperations > 0) {
-                groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise.get(),
-                        Constants.PROGRAMMING_EXERCISE_FAILED_STASH_OPERATIONS_NOTIFICATION + numberOfFailedStashOperations);
-            }
-            else {
-                groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise.get(),
-                        Constants.PROGRAMMING_EXERCISE_SUCCESSFUL_STASH_OPERATION_NOTIFICATION);
-            }
+            var failedStashOperationsFuture = stashChangesInAllStudentRepositories(programmingExerciseId, condition);
+            failedStashOperationsFuture.thenAccept(failedStashOperations -> {
+                long numberOfFailedStashOperations = failedStashOperations.size();
+                if (numberOfFailedStashOperations > 0) {
+                    groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise,
+                            Constants.PROGRAMMING_EXERCISE_FAILED_STASH_OPERATIONS_NOTIFICATION + numberOfFailedStashOperations);
+                }
+                else {
+                    groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise,
+                            Constants.PROGRAMMING_EXERCISE_SUCCESSFUL_STASH_OPERATION_NOTIFICATION);
+                }
```

Comment:
```suggestion
                String notificationText;
                if (numberOfFailedStashOperations > 0) {
                    notificationText = Constants.PROGRAMMING_EXERCISE_FAILED_STASH_OPERATIONS_NOTIFICATION + numberOfFailedStashOperations;
                }
                else {
                    notificationText = Constants.PROGRAMMING_EXERCISE_SUCCESSFUL_STASH_OPERATION_NOTIFICATION;
                }
                groupNotificationService.notifyEditorAndInstructorGroupAboutExerciseUpdate(programmingExercise, notificationText);
```

It's a matter of taste, just take this as a suggestion and choose what you like more.